### DISCUSSION TITLE
Form Functionality for Input Component

### DISCUSSION THREAD

**Discussion by toddhow** [heroui-inc/heroui#744] [Feature requests]
Could form functionality similar to Mantine's be possible in future releases?

**Comment by toddhow** [comment]
https://mantine.dev/form/use-form/

**Comment by matthiasfeist** [comment]
I just saw this thread and wanted to open a similar one myself.
The main reason why I am using Mantine instead of NextUI in my project is actually mainly the DX of how forms are being handled.

I _assume_ that libraries like NextUI and maybe also the currently popular ShadcnUI see themselves more as building blocks. Unfortunately this takes then a lot of boilerplate code to make these libs work together with stuff like ReactFormHook.

One thing that Mantine and maybe also Adobe's React library (see here: https://react-spectrum.adobe.com/react-spectrum/forms.html) do better is that they treat forms and the DX of forms as a first class citizen.

So maybe it would be nice to have some sort of "adapter" or wrapper that would help to integrate NextUI with popular form helpers in a concise and convenient way?
